packages/office-ui-fabric-react/src/components/Sticky/Sticky.tsx

```diff
@@ -98,246 +108,355 @@ export class Sticky extends BaseComponent<IStickyProps, IStickyState> {
-    ) {
-      if (curr.tagName === 'HTML') {
-        // Fallback color if no element has a declared background-color attribute
-        return undefined;
-      }
-      if (curr.parentElement) {
-        curr = curr.parentElement;
-      }
-    }
-    return window.getComputedStyle(curr).getPropertyValue('background-color');
+  return distance;
+}
+
+function _getStickyDistanceFromTopForFooter(
+  container: HTMLElement,
+  footerStickyVisibleContainer: HTMLElement,
+  stickyContentBottom: HTMLDivElement | null
+): number {
+  let distance = 0;
+  if (stickyContentBottom) {
+    distance = container.clientHeight - _getOffsetHeight(footerStickyVisibleContainer) + _getOffsetTop(stickyContentBottom);
   }
+
+  return distance;
 }
 
-function _isOffsetHeightDifferent(a: React.RefObject<HTMLElement>, b: React.RefObject<HTMLDivElement>): boolean {
-  return (a && b && a.current && b.current && a.current.offsetHeight !== b.current.offsetHeight) as boolean;
+function _getStickyPlaceholderHeight(isSticky: boolean, nonStickyContent: HTMLDivElement | null): React.CSSProperties {
+  const height = nonStickyContent ? _getOffsetHeight(nonStickyContent) : 0;
+  return {
+    visibility: isSticky ? 'hidden' : 'visible',
+    height: isSticky ? 0 : height
+  };
+}
+
+function _getContentStyles(
+  isSticky: boolean,
+  optimizePerformance: boolean,
+  root: HTMLDivElement | null,
+  stickyBackgroundColor?: string
+): React.CSSProperties {
+  const isVisible = isSticky ? !optimizePerformance : true;
+  return {
+    backgroundColor: stickyBackgroundColor || _getBackgroundOfNearestAncestor(root),
+    overflow: isSticky ? 'hidden' : '',
+    visibility: isVisible ? 'visible' : 'hidden'
+  };
+}
+
+function _getStickyContent(
+  stickyContentRef: React.RefObject<HTMLDivElement>,
+  nonStickyContent: HTMLDivElement | null,
+  root: HTMLDivElement | null,
+  isSticky: boolean,
+  optimizePerformace: boolean,
+  props: IStickyProps
+): JSX.Element {
+  return (
+    <div ref={stickyContentRef} aria-hidden={!isSticky} style={{ pointerEvents: isSticky ? 'auto' : 'none' }}>
+      {optimizePerformace ? (
+        <div
+          className={isSticky ? props.stickyClassName : undefined}
+          style={_getStickyContentStyles(isSticky, props.stickyBackgroundColor || _getBackgroundOfNearestAncestor(root))}
+        >
+          {props.children}
+        </div>
+      ) : (
+        <div style={_getStickyPlaceholderHeight(isSticky, nonStickyContent)} />
+      )}
+    </div>
+  );
+}
+
+function _getStickyContentStyles(isSticky: boolean, stickyBackgroundColor?: string): React.CSSProperties {
+  return {
+    visibility: isSticky ? 'visible' : 'hidden',
+    pointerEvents: isSticky ? 'auto' : 'none',
+    overflow: 'hidden',
+    backgroundColor: stickyBackgroundColor
+  };
+}
+// debatable
+/**
+ * Returns reactRefObject.current
+ * @param reactRefObject - React.RefObject
+ */
+function _getRefObjectCurrent<T>(reactRefObject: React.RefObject<T>): T | null {
```

Comment:
Same comment as earlier about not making functions just for property access

Reply:
string1=`'function xn(e){return e.current}xn()xn()xn()xn()xn()xn()xn()'`
string2=`'.current.current.current.current.current.current.current'`
string1.length-string2.length=4, but in future if .current is read more times, it would contribute 4B instead of 8B

Even after adding new code, Sticky.min.js size is reduced by 236B.

![image](https://user-images.githubusercontent.com/41147089/67371263-bf4ff300-f599-11e9-9c28-3d3234f54629.png)